_inc/lib/core-api/wpcom-endpoints/publicize-connections.php

```diff
@@ -0,0 +1,163 @@
+<?php
+
+/**
+ * Publicize: List Connections
+ *
+ * [
+ *   { # Connnection Object. See schema for more detail.
+ *     id:           (string)  Connection unique_id
+ *     service_name: (string)  Service slug
+ *     display_name: (string)  User name/display name of user/connection on Service
+ *     global:       (boolean) Is the Connection available to all users of the site?
+ *   },
+ *   ...
+ * ]
+ * @since 6.8
+ */
+class WPCOM_REST_API_V2_Endpoint_List_Publicize_Connections extends WP_REST_Controller {
+	public function __construct() {
+		$this->namespace = 'wpcom/v2';
+		$this->rest_base = 'publicize/connections';
+
+		add_action( 'rest_api_init', array( $this, 'register_routes' ) );
+	}
+
+	/**
+	 * Called automatically on `rest_api_init()`.
+	 */
+	public function register_routes() {
+		register_rest_route( $this->namespace, '/' . $this->rest_base, array(
+			array(
+				'methods'             => WP_REST_Server::READABLE,
+				'callback'            => array( $this, 'get_items' ),
+				'permission_callback' => array( $this, 'get_items_permission_check' ),
+			),
+			'schema' => array( $this, 'get_public_item_schema' ),
+		) );
+	}
+
+	/**
+	 * Helper for generating schema. Used by this endpoint and by the
+	 * Connection Test Result endpoint.
+	 *
+	 * @internal
+	 */
+	protected function get_connection_schema_properties() {
+		return array(
+			'id' => array(
+				'description' => __( 'Unique identifier for the Publicize Connection', 'jetpack' ),
+				'type' => 'string',
+			),
+			'service_name' => array(
+				'description' => __( 'Alphanumeric identifier for the Publicize Service', 'jetpack' ),
+				'type' => 'string',
+			),
+			'display_name' => array(
+				'description' => __( 'Username of the connected account', 'jetpack' ),
+				'type' => 'string',
+			),
+			'global' => array(
+				'description' => __( 'Is this connection available to all users?', 'jetpack' ),
+				'type' => 'boolean',
+			),
+		);
+	}
+
+	public function get_item_schema() {
+		$schema = array(
+			'$schema' => 'http://json-schema.org/draft-04/schema#',
+			'title' => 'jetpack-publicize-connection',
+			'type' => 'object',
+			'properties' => $this->get_connection_schema_properties(),
+		);
+
+		return $this->add_additional_fields_schema( $schema );
+	}
+
+	/**
+	 * Helper for retrieving Connections. Used by this endpoint and by
+	 * the Connection Test Result endpoint.
+	 *
+	 * @internal
+	 */
+	protected function get_connections() {
+		global $publicize;
+
+		$items = array();
+
+		foreach ( (array) $publicize->get_services( 'connected' ) as $service_name => $connections ) {
+			foreach ( $connections as $connection ) {
+				$connection_meta = $publicize->get_connection_meta( $connection );
+				$connection_data = $connection_meta['connection_data'];
+
+				$items[] = array(
+					'id' => $publicize->get_connection_unique_id( $connection ),
```

Comment:
Do we need to cast to string here to ensure that the ID will be string?

Reply:
Since we've defined the schema, I *think* it's handled by core, but I agree we should do it here anyway.

In theory, `Publicize::get_connection_unique_id()` should be returning a string anyway, but… I suspect it's not. We should probably add casting upstream.

Fixed here in 500e82d87.